app/models/custom_attributes/custom_attribute_definition.rb

```diff
@@ -4,8 +4,14 @@ module CustomAttributeDefinition
     included do
       scope :by_sort_order, -> { order('sort_order ASC') }
       scope :visible_to_staff, -> { where(hide_visibility_from_staff: false) }
+      scope :datebox_attributes, -> { where(attr_type: 'date') }
       validates :attr_name, presence: true, format: { with: /^[a-zA-Z\_\s]+$/, multiline: true, message: 'cannot contain special characters.' }
       validates :default_value, format: { with: /^[\+\-]?\d*\.?\d*$/, multiline: true, message: 'should be a number.' }, if: :number_type?
+      validate :validate_custom_attr_date_alert_options
+      before_update :recurring_custom_attribute_previous
```

Comment:
no need to do it in this way
read and understand `https://api.rubyonrails.org/classes/ActiveModel/Dirty.html` and then use `changed?` or `previously_changed?`